addOns/client/src/main/java/org/zaproxy/addon/client/RedirectScript.java

```diff
@@ -38,5 +38,8 @@ public void browserLaunched(SeleniumScriptUtils ssutils) {
         JavascriptExecutor jsExecutor = (JavascriptExecutor) ssutils.getWebDriver();
         jsExecutor.executeScript("localStorage.setItem('localzapurl', '" + zapurl + "')");
         jsExecutor.executeScript("localStorage.setItem('localzapenable',false)");
+
+        // This statement make sure that the ZAP browser extension is configured properly
+        ssutils.getWebDriver().get(zapurl);
```

Comment:
still there is a chance for race condition

Reply:
I think it would be better doing it this way. It is more cleaner and safer as well.

Reply:
Hmm, can't we wait in the script for everything to be set before reloading?

Reply:
I don't know about that. Might get complicated as well. 🤔

Reply:
How so, we do the same in the browser extension, the context is the same.

Reply:
Then we will need to have check to check weather the domain is zap domain or not before reloading. It is possible but I think things will get complicated unnecessarily. Do you think there can be any problem with the current approach?

Reply:
It can only be the zap domain that was what we accessed in the previous statement. The current approach is a workaround (IMO), since we are just trying to delay the local storage access. Not a blocker, we can tweak this in following releases.

Reply:
Browser extension will access the local storage two times. First time when we go to `zapurl` we set the local storage variables, in the meantime browser extension is also fetching those variables. But in the second time the local storage is already set so the browser extension grabs all the set local storage variables.

Reply:
I know that's the objective, but still not see why doing that in the script all at once would not work (better do that "atomically").